saleor/graphql/order/mutations/orders.py

```diff
@@ -606,36 +620,85 @@ def perform_mutation(cls, _root, info, amount, **data):
                 }
             )
 
+    @classmethod
+    def _check_amount_to_refund(cls, payments, amount=None):
+        if amount:
+            total_captured_amount = sum(
+                [item["payment"].captured_amount for item in payments]
+            )
+            if amount > total_captured_amount:
+                raise ValidationError(
+                    {
+                        "amount": ValidationError(
+                            "The total amount to refund cannot be bigger "
+                            "than the total captured amount.",
+                            code=OrderErrorCode.AMOUNT_TO_REFUND_TOO_BIG,
+                        )
+                    }
+                )
+
+        else:
+            improper_payments_ids = []
+            for item in payments:
+                if item["payment"].captured_amount < item["amount"]:
+                    improper_payments_ids.append(
+                        to_global_id("Payment", item["payment"].id)
+                    )
+
+            if improper_payments_ids:
+                raise ValidationError(
+                    {
+                        "amount": ValidationError(
+                            "The amount to refund cannot be bigger "
+                            "than the captured amount.",
+                            code=OrderErrorCode.AMOUNT_TO_REFUND_TOO_BIG,
+                            params={"improper_payments_ids": improper_payments_ids},
```

Comment:
You should also update `OrderError` to attach the ids to the error. I will also suggest to rename the field to just `payments` and it's done like that in different errors.

After updating this you should also check the returning IDs in tests.